utils/aarch64_32_bit_el1_supported.c

```diff
@@ -0,0 +1,35 @@
+#include <errno.h>	/* errno for perror() */
+#include <fcntl.h>	/* open() */
+#include <stdio.h>	/* perror() */
+#include <sys/ioctl.h>	/* _IO(), ioctl() */
+#include <unistd.h>	/* close() */
+
+/* From include/uapi/linux/kvm.h */
+#define KVM_CHECK_EXTENSION	_IO(0xAE, 0x03)
+#define KVM_CAP_ARM_EL1_32BIT	93
+
+int main(void)
+{
+	int fd, ret;
+
+	fd = open("/dev/kvm", O_RDWR);
+	if (fd < 0) {
+		perror("Failed to open /dev/kvm");
+		return -errno;
+	}
+
+	ret = ioctl(fd, KVM_CHECK_EXTENSION, KVM_CAP_ARM_EL1_32BIT);
+	if (ret < 0) {
+		perror("Error checking /dev/kvm for 32-bit guest support");
+	} else {
+		/*
+		 * KVM_CHECK_EXTENSION returns 1 for supported, 0 for
+		 * unsupported so invert it to match typical success/fail codes
+		 * in programs.
+		 */
+		ret = !ret;
+	}
+
+	close(fd);
+	return ret;
```

Comment:
```c
if (ret < 0) {
  perror(...);
  ret = 0;
}

close(fd);
return !ret;
```

Reply:
Thanks, changed. I kept the comment, yell if you disagree.